src/main/java/org/apache/mesos/scheduler/plan/DefaultPlanManager.java

```diff
@@ -38,235 +19,14 @@ public Plan getPlan() {
-
-        return null;
-    }
-
-    @Override
-    public Status getStatus() {
-        return PlanManagerUtils.getStatus(getPlan(), phaseStrategies);
-    }
-
-    @Override
-    public Status getPhaseStatus(final UUID phaseId) {
-        PhaseStrategy strategy = getStrategy(getPhase(phaseId));
-        return strategy != null ? strategy.getStatus() : Status.ERROR;
-    }
-
     @Override
-    public List<String> getErrors() {
-        return plan.getErrors();
-    }
-
-    @Override
-    public void update(Observable o, Object obj) {
-        if (obj instanceof Protos.TaskStatus) {
-            update((Protos.TaskStatus) obj);
-        }
-    }
-
-    private Optional<PhaseStrategy> getCurrentPhaseStrategy() {
-        Optional<Phase> phaseOptional = getCurrentPhase();
-        return phaseOptional.isPresent() ? Optional.of(getStrategy(phaseOptional.get())) : Optional.empty();
-    }
-
-    /**
-     * Returns the matching {@link PhaseStrategy} for the provided {@code phaseId}/{@code blockId}.
-     */
-    private PhaseStrategy getPhaseStrategy(final UUID phaseId, final UUID blockId) {
-        Phase phase = getPhase(phaseId);
-        if (phase == null) {
-            LOGGER.info("Phase not found : phaseId = {}", phaseId);
-            return null;
-        }
-        Block block = phase.getBlock(blockId);
-        if (block == null) {
-            LOGGER.info("Block not found: phaseId =  {}, blockId = {}",
-                    phaseId,
-                    blockId);
-            return null;
-        }
-        PhaseStrategy strategy = getStrategy(phase);
-        if (strategy == null) {
-            LOGGER.info("Strategy not found: phaseId = {}, blockId = {}",
-                    phaseId,
-                    blockId);
-            return null;
-        }
-        return strategy;
-    }
-
-    /**
-     * Returns a matching {@link PhaseStrategy} for the provided {@link Phase}.
-     * If the {@link PhaseStrategy} doesn't exist yet, it is created using {@link PhaseStrategyFactory}.
-     */
-    protected PhaseStrategy getStrategy(final Phase phase) {
-        if (phase == null) {
-            LOGGER.warn("null phase");
-            return null;
-        }
-        return phaseStrategies.get(phase.getId());
-    }
-
-    /**
-     * Returns the {@link Phase} which has the provided {@code phaseId}, or {@code null} if no
-     * matching {@link Phase} was found.
-     */
-    protected Phase getPhase(final UUID phaseId) {
-        if (phaseId == null) {
-            LOGGER.warn("null phaseId");
-            return null;
-        }
-
-        for (Phase phase : plan.getPhases()) {
-            if (phaseId.equals(phase.getId())) {
-                return phase;
-            }
-        }
+    public Collection<? extends Block> getCandidates(Collection<String> dirtyAssets) {
+        Collection<Phase> candidatePhases = plan.getStrategy().getCandidates(plan, dirtyAssets);
+        Collection<Block> candidateBlocks = candidatePhases.stream()
```

Comment:
Doesn't this break the execution order amongst phases ? This will work just fine for a single phase and multiple parallel phases. But, will break the execution order when serial phase execution is desired ? OR this relies on the underlying `PhaseStrategy` to do the right thing ?